src/System.Management.Automation/engine/parser/TypeInferenceVisitor.cs

```diff
@@ -2858,98 +2855,423 @@ private static CommandBaseAst GetPreviousPipelineCommand(CommandAst commandAst)
             var i = pipe.PipelineElements.IndexOf(commandAst);
             return i != 0 ? pipe.PipelineElements[i - 1] : null;
         }
-    }
 
-    internal static class TypeInferenceExtension
-    {
-        public static bool EqualsOrdinalIgnoreCase(this string s, string t)
+        private sealed class VariableAssignmentVisitor : AstVisitor2
         {
-            return string.Equals(s, t, StringComparison.OrdinalIgnoreCase);
-        }
+            internal bool LocalScopeOnly;
+            internal bool ScopeIsLocal;
+            /// <summary>
+            /// The variable that we are trying to determine the type of.
+            /// </summary>
+            internal VariableExpressionAst VariableTarget;
+            internal int StopSearchOffset;
+            /// <summary>
+            /// The last type constraint applied to the variable. This takes priority when determining the type of the variable.
+            /// </summary>
```

Comment:
Please add blank line doc comments here and below:
```suggestion
            internal int StopSearchOffset;

            /// <summary>
            /// The last type constraint applied to the variable. This takes priority when determining the type of the variable.
            /// </summary>
```